Feat summary per class script

i-chvets opened the pull request:
CVE scanner script produces a report with counts of CVEs per severity. Trivy report provide Class of the target for which CVE was detected. There are two classes `os-pkgs` and `lang-pkgs`. This PR contains changes that generate CVE report per class.
This will allow more accurate analysis of CVEs.

Summary of changes:
- Added script that generates CVE count summary per class, i.e. for OS and LANG packages classes in Trivy reports.
- Modified main shell script to properly format report and call new script.
NOTE: When scanner cannot identify base image it is indicated as `N/A`. This is the case for ROCK images.

Testing new format:
```
IMAGE,BASE,CRITICAL,HIGH,MEDIUM,LOW,CRITICAL-OS,CRITICAL-LANG,HIGH-OS,HIGH-LANG,MEDIUM-OS,MEDIUM-LANG,LOW-OS,LOW-LANG
docker.io/charmedkubeflow/seldon-core-operator:v1.15.0_22.04_1,N/A,0,4,0,0,0,0,0,4,0,0,0,0
docker.io/kubeflownotebookswg/volumes-web-app:v1.7.0,debian:11.6,1,9,10,0,0,1,8,1,8,2,0,0
docker.io/kubeflownotebookswg/notebook-controller:v1.7.0,debian:11.6,0,58,21,0,0,0,4,54,4,17,0,0
kubeflownotebookswg/tensorboard-controller:v1.7.0,debian:11.6,0,6,1,0,0,0,0,6,0,1,0,0
docker.io/kubeflownotebookswg/kfam:v1.7.0,debian:11.6,0,15,8,0,0,0,4,11,4,4,0,0
kubeflownotebookswg/tensorboards-web-app:v1.7.0,debian:11.6,1,9,10,0,0,1,8,1,8,2,0,0
docker.io/kubeflownotebookswg/jupyter-web-app:v1.7.0,debian:11.6,1,9,10,0,0,1,8,1,8,2,0,0
docker.io/charmedkubeflow/mlserver-xgboost:1.2.0_22.04_1,ubuntu:22.04,3,8,5,4,0,3,0,8,0,5,0,4
docker.io/seldonio/engine:1.12.0,debian:10.2,36,95,95,6,21,15,56,39,50,45,4,2
docker.io/seldonio/seldon-core-executor:1.14.0,debian:10.12,6,17,16,2,6,0,14,3,16,0,2,0
docker.io/charmedkubeflow/mlserver-mlflow:1.2.0_22.04_1,ubuntu:22.04,4,9,6,4,0,4,0,9,0,6,0,4
docker.io/kubeflownotebookswg/centraldashboard:v1.7.0,alpine:3.15.4,6,29,20,2,1,5,10,19,10,10,0,2
docker.io/charmedkubeflow/mlserver-sklearn:1.2.0_22.04_1,ubuntu:22.04,3,8,5,4,0,3,0,8,0,5,0,4
docker.io/kubeflownotebookswg/profile-controller:v1.7.0,debian:11.6,1,56,21,0,0,1,4,52,4,17,0,0
```

kimwnasptd commented:
@i-chvets I had couple of review comments but ended up writing a lot of code. I've prepared the branch `feat-summary-per-class-script-rev` and pushed 2 commits there.

The high level overview of my changes being:
1. I exposed information about all the CVEs as well, so that we can have the wholistic overview first and then the breakdown to os and lang package CVEs
2. Broke down the code into functions to (try to) make it more readable
3. Kept all code under 80 chars per line

PTAL

i-chvets commented:
@kimwnasptd Your version looks good. Let's get PR going. However I will need to do minor change to shell scan images script to use this new script and then we will have a complete reporting tool.

kimwnasptd commented:
ACK! Feel free of course to cherry-pick the commits from the review branch. Give me a heads up once you've made the changes you need and I can take another look

i-chvets commented:
> ACK! Feel free of course to cherry-pick the commits from the review branch. Give me a heads up once you've made the changes you need and I can take another look

Pulled your changes in and did tons of linting. Updated shell script, much cleaner now. Updated sample format in PR description. Also `get-summary.py` can be used on its own to generate CSV file.
Ready to be approved/merged.

kimwnasptd reviewed (approved):
Looks good!

kimwnasptd commented on scripts/images/scan-images.sh:
```diff
@@ -22,7 +21,7 @@ DATE=$(date +%F)
 if [ ! -f $SCAN_SUMMARY_FILE ]; then
     # create header for scan summary file, if it does not exist
-    echo "IMAGE,BASE,CRITICAL,HIGH,MEDIUM,LOW " >> $SCAN_SUMMARY_FILE
+    echo "IMAGE,BASE,OS-CRITICAL,OS-HIGH,OS-MEDIUM,OS-LOW,LANG-CRITICAL,LANG-HIGH,LANG-MEDIUM,LANG-LOW" >> $SCAN_SUMMARY_FILE
```
Oops, was from when I started the review. Resolving

i-chvets replied:
Will do.

i-chvets replied:
Yes. It is all updated and CVEs header has titels that group by OS and LANG.
Merging it.
